.github/workflows/push-release-build.yaml

```diff
@@ -0,0 +1,56 @@
+name: Push Release Image Build
+on:
+  pull_request:
+    branches:
+      - release-*
+      - master
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+jobs:
+  push-release-image-to-container-registry:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+       # docker/setup-qemu action installs QEMU static binaries, which are used to run builders for architectures other than the host.
+      - name: set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
+
+      - name: log in to container registry
+        uses: docker/login-action@v1
+        with:
+          username: ""
+          password: ""
+
+      # - name: Configure AWS Credentials
+      #   uses: aws-actions/configure-aws-credentials@v1
+      #   with:
+      #     aws-access-key-id: ${{ secrets.AWS_USR }}
+      #     aws-secret-access-key: ${{ secrets.AWS_PSW }}
+      #     aws-region: us-east-1
+
+      - name: unshallow
+        run: git fetch --prune --unshallow --tags
+
+      # creating custom env var
+      # - name: set env
+      #   run: echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: build and release
+        env:
+          GIT_API_TOKEN: ${{ secrets.GIT_API_TOKEN }}
```

Comment:
even after putting some random creds for docker hub...it will still require secrets for GitHub token. I'm closing this, I'll have to think of another way for testing.